src/gui/raster/qgsrasterlayerproperties.h

```diff
@@ -85,12 +87,40 @@ class GUI_EXPORT QgsRasterLayerProperties : public QgsOptionsDialogBase, private
      */
     void addPropertiesPageFactory( const QgsMapLayerConfigWidgetFactory *factory );
 
+    QgsExpressionContext createExpressionContext() const override;
+
   protected slots:
     //! \brief auto slot executed when the active page in the main widget stack is changed
     void optionsStackedWidget_CurrentChanged( int index ) override SIP_SKIP ;
 
+  private:
+
+    // TODO -- consider moving these to a common raster widget base class
+
+    /**
+     * Registers a property override button, setting up its initial value, connections and description.
+     * \param button button to register
+     * \param key corresponding data defined property key
```

Comment:
\since QGIS 3.22

Reply:
Ah, it's private methods, but I've added it anyway